server/routerlicious/packages/routerlicious-base/src/utils/sessionHelper.ts

```diff
@@ -287,7 +288,35 @@ export async function getSession(
 ): Promise<ISession> {
 	const baseLumberjackProperties = getLumberBaseProperties(documentId, tenantId);
 
-	const document = await documentRepository.readOne({ tenantId, documentId });
+	let document;
+	try {
+		// Retry document existence check to avoid document DB race condition
+		document = await runWithRetry(
+			async () =>
+				documentRepository.readOne({ tenantId, documentId }).then((result) => {
+					if (result === null) {
+						throw new NetworkError(
+							404,
+							"Document is deleted and cannot be accessed",
+							true /* canRetry */,
+						);
+					}
+					return result;
+				}),
+			"getDocumentForSession",
+			2, // maxRetries
+			500, // retryAfterMs
+			baseLumberjackProperties, // telemetry props
+			undefined,
+			(error) => shouldRetryNetworkError(error),
+		);
+	} catch (error) {
+		// Add a stage stamp before throwing the error
+		connectionTrace?.stampStage("DocumentDoesNotExist");
```

Comment:
This should probably only be `if (isNetworkError(error) && error.code === 404)`. I assume there could be some other errors like DB errors or something that don't mean DocumentDoesNotExist

Reply:
Fixed as part of other change